pyxrf/model/lineplot.py

```diff
@@ -264,6 +264,39 @@ def _update_energy(self, change):
             return
         self.incident_energy = self.parameters['coherent_sct_energy']['value']
 
+    def set_incident_energy(self, change):
+        """
+        The observer function that changes the value of incident energy
+        and upper bound for fitted energy range.
+
+        Parameters
+        ----------
+
+        energy_new : float
+            New value of incident energy
+        """
+
+        margin = 0.8  # Value by which the upper bound of the range used for fitting
+        #               exceeds the incident energy. Selected for convenience, but
+        #               is subject to change. This is the place to change it to take effect
+        #               throughout the program.
+
+        energy_new = change["value"]
+        # Limit the number of decimal points for better visual presentation
+        energy_new = float(f"{energy_new:.5f}")
```

Comment:
I replaced it with ``np.round``, but it needs to be tested.

Reply:
Is it an array? I don't think we need to overcomplicate it in case of a single number, so a native Python `round` function will just work:
```py
In [6]: round(1.2345678, 5)
Out[6]: 1.23457
```

Reply:
Fixed